### DISCUSSION TITLE
Windows visual studio

### DISCUSSION THREAD

**Discussion by Ilyzxc** [reo7sp/tgbot-cpp#198] [Q&A] [answered]
How to add this library to Visual studio 2022 on Windows OS?

**Comment by nekotxs** [comment]
you can install it by vcpkg or add this lib as submodule

**Comment by reo7sp** [comment, marked as answer]
Current tgbot-cpp requires Visual Studio 2022 with the Desktop C++ workload, CMake, Ninja, Make and Conan 2. The exact installation commands are in the Windows section of the repository README; run the Make targets from Git Bash.

Build and install the library:

```sh
make install INSTALL_PREFIX=build/install
```

In the bot project, use CMake targets instead of adding `.lib` files manually:

```cmake
find_package(TgBot CONFIG REQUIRED)
add_executable(your_bot main.cpp)
target_link_libraries(your_bot PRIVATE TgBot::TgBot)
```

Configure the consumer with the same Conan-generated toolchain and add the tgbot-cpp install prefix to `CMAKE_PREFIX_PATH`. An alternative is to add tgbot-cpp as a Git submodule and use `add_subdirectory`; the repository contains a complete `echobot-submodule` example.
